include/zephyr/drivers/sensor_types.h

```diff
@@ -0,0 +1,312 @@
+/**
+ * @file drivers/sensor_types.h
+ *
+ * @brief Public APIs for the sensor driver.
+ */
+
+/*
+ * Copyright (c) 2022 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef INCLUDE_DRIVERS_SENSOR_TYPES_H_
+#define INCLUDE_DRIVERS_SENSOR_TYPES_H_
+
+#include <math/util.h>
+
+#ifdef CONFIG_CHRE
+#define CHRE_VALUE_ASSERT(name) BUILD_ASSERT(name == _CONCAT(CHRE_, name))
+#else
+#define CHRE_VALUE_ASSERT(name)
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file
+ * The SENSOR_TYPE_* defines are the sensor types supported. Additional types can be added at
+ * SENSOR_TYPE_VENDOR_START.
+ */
+
+/**
+ * Start value for all of the vendor-defined private sensors.
+ */
+#define SENSOR_TYPE_VENDOR_START UINT8_C(192)
+CHRE_VALUE_ASSERT(SENSOR_TYPE_VENDOR_START);
```

Comment:
All of these CHRE asserts should be moved to the CHRE integration layer.

Reply:
That will require an upstream change to the zephyr integration. Can we leave that as a TODO? The ordering will need to be:
1. Merge this PR
2. Update upstream to add checks (since these constants don't exist yet)
3. Pull those changes to our fork
4. Update west.yaml

Reply:
No. Maybe you could put them in an in-tree include file in `/modules/chre`, but I don't want to introduce a major driver API change with an apparent dependency on CHRE.

Reply:
Done